src/ppx/grfx/grfx_swapchain.cpp

```diff
@@ -155,20 +202,46 @@ void Swapchain::DestroyColorImages()
 
 Result Swapchain::CreateDepthImages()
 {
-    if ((mCreateInfo.depthFormat != grfx::FORMAT_UNDEFINED) && mDepthImages.empty()) {
-        for (uint32_t i = 0; i < mCreateInfo.imageCount; ++i) {
-            grfx::ImageCreateInfo dpCreateInfo = ImageCreateInfo::DepthStencilTarget(mCreateInfo.width, mCreateInfo.height, mCreateInfo.depthFormat);
-            dpCreateInfo.ownership             = grfx::OWNERSHIP_RESTRICTED;
-            dpCreateInfo.DSVClearValue         = {1.0f, 0xFF};
+    if (!mDepthImages.empty()) {
+        return ppx::SUCCESS;
+    }
+    if (mCreateInfo.depthFormat == grfx::FORMAT_UNDEFINED) {
+        return ppx::SUCCESS;
+    }
+    for (uint32_t i = 0; i < mCreateInfo.imageCount; ++i) {
+        grfx::ImageCreateInfo dpCreateInfo = ImageCreateInfo::DepthStencilTarget(mCreateInfo.width, mCreateInfo.height, mCreateInfo.depthFormat);
+        dpCreateInfo.ownership             = grfx::OWNERSHIP_RESTRICTED;
+        dpCreateInfo.DSVClearValue         = {1.0f, 0xFF};
```

Comment:
This is already the default value. Can be removed.

Reply:
slightly different opinion here, only for discussion. :) personally, i feel it is better to assign the value here even it is using the default value so that it is clear what value is assigned to the member, otherwise, the user need to go to the definition of the struct and look for the default values.